### DISCUSSION TITLE
Remove redundant headers

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#800] [General]
Headers are useful to improve readability of the code, and to logically group functions between it's visibility and it's type (constant/non-constant).

But, there are places where, IMO, where these headers are a chore (and actually decrease the readability).

E.g. different headers between constants and storage variables, and internal vs private:
https://github.com/sablier-labs/v2-core/blob/7819a3b5122036596edff3326b97a8c03750bbb8/src/abstracts/SablierV2Base.sol#L24-L39

My proposal is to use only one header for these variables, and that is:
```solidity
    /*//////////////////////////////////////////////////////////////////////////
                                 CONTRACT STATE
    //////////////////////////////////////////////////////////////////////////*/
```

wdyt? @PaulRBerg  @smol-ninja

**Comment by scorpion9979** [comment]
@andreivladbrg When discussing changes that don't impact code performance, I'm of the opinion of optimizing code readability for auditors. I think categorized headers could be further made valuable when supercharged with region folding. Region comments in Solidity are already supported by [juanfranblanco's VSCode extension](https://github.com/juanfranblanco/vscode-solidity), but I've been trying to [have Hardhat support it as well](https://github.com/NomicFoundation/hardhat-vscode/pull/526).

**Reply by andreivladbrg** [reply]
thank you for your answer, and i am sorry for the random tag, it was mistake

**Reply by scorpion9979** [reply]
@andreivladbrg We're good! No need to apologize :)

**Comment by PaulRBerg** [comment]
Interesting proposal.

I am not sure, however, if what you mean is get rid of the public<>internal dichotomy for storage variables only? Or do you want that for functions, as well?

**Reply by andreivladbrg** [reply]
>  if what you mean is get rid of the public<>internal dichotomy for storage variables only? 

yes, only for storage/constant variables, and **not** for functions

functions can remain the same

**Reply by PaulRBerg** [reply]
Ok in this case, I am ok with the proposal.

**Reply by smol-ninja** [reply]
@andreivladbrg I like the proposal and I do agree with you that having these many headers for storage decreases the readability sometimes. However, I would love to have some more clarity on the OP. 

We have used the following headers:

- Public constants
- Internal constants
- Public storage
- Private storage
- Internal storage

Do you plan to use CONTRACT STATE for all of them? If yes, I think it would be a good idea to not use it for constants because technically they are not contract states. We can have CONSTANTS (to cover internal, private and public) and CONTRACT STATE (to cover private, public and internal).

**Reply by PaulRBerg** [reply]
Good point, and I agree that using `CONSTANTS` and `CONTRACT STATE` is better.

Or maybe `STATE VARIABLES` instead.

**Reply by smol-ninja** [reply]
I like `STATE VARIABLES` over `CONTRACT STATE`.

**Reply by andreivladbrg** [reply]
I precisely said "State" to refer to both constants and storage variables which _is correct_ and **not** confusing in context of EVM. 

As we can see here: https://takenobu-hs.github.io/downloads/ethereum_evm_illustrated.pdf 
There are two kinds of accounts, EOAs and contracts.

Contracts accounts' state is both storage and code:
<img width="722" alt="image" src="https://github.com/sablier-labs/v2-core/assets/99738872/26c8b589-e2bc-4ab7-9e70-1b11120052d5">


Thus, we should use `CONTRACT STATE` for both

**Reply by smol-ninja** [reply]
I must say you made a very interesting point by referring to the definition of `CONTRACT STATE` in the context of EVM. However, as I was thinking about it, I wondered if we should use terminologies in the context of EVM or solidity language.

For example, in the context of Solidity, state variables are used for both storage variables as well as constants.

1. [State variables are variables whose values are permanently stored in contract storage.](https://docs.soliditylang.org/en/v0.8.23/structure-of-a-contract.html)
2. [State variables can be declared as constant or immutable.](https://docs.soliditylang.org/en/v0.8.23/contracts.html#constant-and-immutable-state-variables)

I am not saying that we should use `STATE VARIABLES` instead of `CONTRACT STATE` but out of curiosity, what do you think about this? 

PS: State variables (constants + storage variables) appears to be a subset of contract state (nonce, balance, storage variables, deployed code).

**Reply by andreivladbrg** [reply]
> I wondered if we should use terminologies in the context of EVM or solidity language

Tbh, this sentence doesn't make sense to me. Isn't Solidity the language that targets the EVM😅?

For me, it doesn't matter if it is either `CONTRACT STATE` or `STATE VARIABLES`, both work.

My response was just against the idea that we should have two different headers for storage and constants, when we can refer to both, as _state_.


Basically responding to these.
Shub:
>> Do you plan to use CONTRACT STATE for all of them? If yes, I think it would be a good idea to not use it for constants because technically they are not contract states. We can have CONSTANTS (to cover internal, private and public) and CONTRACT STATE (to cover private, public and internal).


PRB:
>> I agree that using CONSTANTS and CONTRACT STATE is better.

**Reply by smol-ninja** [reply]
I agree with you. And I was wrong to say that constants are not contract states. They indeed are.

**Reply by PaulRBerg** [reply]
> I wondered if we should use terminologies in the context of EVM or solidity language

This sentence does make sense to me.

> Isn't Solidity the language that targets the EVM😅?

It does target the EVM, but it can contain a higher-level concept that has nothing to do (directly) with raw EVM bytecode.

---

I suggest making a PR to see what the new headers look like in practice.

We can make a decision on the spot - whether we like it without `CONSTANTS`, or not.

**Comment by PaulRBerg** [comment]
Also FYI @andreivladbrg, we don't apply labels to discussions.
